src/components/ActionsMenuLayout/ActionsMenuLayout.tsx

```diff
@@ -16,30 +20,98 @@ export class ActionsMenuLayout extends React.Component<ActionsMenuLayoutProps> {
   static Divider = ActionsMenuLayoutDivider;
 
   static displayName = 'ActionsMenuLayout';
+  state = {
+    focusedIdx: 0,
+  };
+
+  liRefs = [];
+
+  componentDidMount() {
+    const { focusedIndex } = this.props;
+
+    if (focusedIndex) {
+      this.focusItem(focusedIndex);
+    } else {
+      this.focusItem(this.state.focusedIdx);
+    }
+  }
+
+  focusItem(idx: number) {
+    this.setState(
+      {
+        focusedIdx: idx,
+      },
+      () => {
+        this.liRefs[idx].liRef?.current.focus();
+      },
+    );
+  }
 
   getDataAttributes(mobile: boolean) {
     return {
       [ACTIONS_MENU_DATA_KEYS.mobile]: mobile,
     };
   }
 
+  getNextItemIdx(direction: 1 | -1): number {
+    const nextItem = this.state.focusedIdx + direction;
+
+    if (direction === 1) {
+      return nextItem === this.liRefs.length ? 0 : nextItem;
+    }
+
+    return nextItem < 0 ? this.liRefs.length - 1 : nextItem;
+  }
+
+  _onKeyDown = (e: React.KeyboardEvent<HTMLUListElement>) => {
+    if (e.key === KEYS.ArrowDown || e.key === KEYS.ArrowDownIE) {
+      e.preventDefault(); // remove scroll on space
+      this.focusItem(this.getNextItemIdx(1));
+    }
+
+    if (e.key === KEYS.ArrowUp || e.key === KEYS.ArrowUpIE) {
+      e.preventDefault(); // remove scroll on space
+      this.focusItem(this.getNextItemIdx(-1));
+    }
+  };
+
+  addRef(ref, index: number) {
+    if (ref) {
+      if (
+        this.liRefs.find(_ref => {
+          return _ref?.props.content === ref.props.content;
+        })
+      ) {
+        return;
+      }
+
+      this.liRefs[
+        index === 0 || this.liRefs[index - 1] ? index : index - 1
+      ] = ref;
```

Comment:
Not sure I follow this part.
Thinking out loud: would it be easier to use a map instead of an array?
So that `this.liRefs` will look like this:
```js
{
  '0': firstRef,
  '1': secondRef,
  ...
}
```
and then just have 
```js
this.liRefs[index] = ref
```

Reply:
I guess I can play around with it. But the `children` of `Layout` component can be a `divider` also. 
So, the map will be something like this:
```
{
  '0': firstRef,
  '1': secondRef,
  '3': secondRef,
  ...
}
```

and we should convert maybe to normalized array with `Object.values`. Wdyt?

Reply:
But maybe with map code will be easier to read. will try to change it and play around with

Reply:
yeah, I forgot about the divider 🤦